kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -132,12 +134,31 @@ public Snapshot getSnapshotAt(
             long version) throws TableNotFoundException {
 
         Optional<LogSegment> logSegmentOpt = getLogSegmentForVersion(engine,
-            Optional.empty(), /* startCheckpointOpt */
-            Optional.of(version) /* versionToLoadOpt */);
-
-        return logSegmentOpt
-            .map(logSegment -> createSnapshot(logSegment, engine))
-            .orElseThrow(() -> new TableNotFoundException(tablePath.toString()));
+                Optional.empty(), /* startCheckpointOpt */
```

Comment:
Add the details of steps to load a snapshot at the given version.
1) for non-coordinated commit table
2) for coordinated commit table.

Reply:
Basically convey the point that for non-coordinated tables this change is a no-op